api/tools/src/main/java/org/pmiops/workbench/tools/CreateTerraMethodSnapshot.java

```diff
@@ -0,0 +1,210 @@
+    return (args) -> {
+      CommandLine opts = new DefaultParser().parse(options, args);
+
+      String configJsonFilepath = opts.getOptionValue(configJsonOpt.getLongOpt());
+      String sourceGitRepo = opts.getOptionValue(sourceGitRepoOpt.getLongOpt());
+      String sourceGitPath = opts.getOptionValue(sourceGitPathOpt.getLongOpt());
+      String sourceGitRef = opts.getOptionValue(sourceGitRefOpt.getLongOpt());
+      String methodNamespace = opts.getOptionValue(methodNamespaceOpt.getLongOpt());
+      String methodName = opts.getOptionValue(methodNameOpt.getLongOpt());
+
+      WorkbenchConfig workbenchConfig =
+          CreateWgsCohortExtractionBillingProjectWorkspace.workbenchConfig(configJsonFilepath);
+      ApiClientFactory apiClientFactory =
+          CreateWgsCohortExtractionBillingProjectWorkspace
+              .wgsCohortExtractionServiceAccountApiClientFactory(workbenchConfig);
+
+      String sourceFileContents = getGithubFileContents(sourceGitRepo, sourceGitPath, sourceGitRef);
+
+      List<FirecloudMethodResponse> existingMethods =
+          apiClientFactory
+              .methodRepositoryApi()
+              .listMethodRepositoryMethods(
+                  methodNamespace, methodName, null, null, null, null, null, null, null);
+
+      DateFormat formatter = new SimpleDateFormat("EEE MMM dd HH:mm:ss zzz yyyy");
+      formatter.setTimeZone(TimeZone.getTimeZone("America/New_York"));
+
+      String snapshotComment =
+          "Sourced from Github (repo, ref, path) ("
+              + sourceGitRepo
+              + ", "
+              + sourceGitRef
+              + ", "
+              + sourceGitPath
+              + ") on "
+              + formatter.format(new Date());
+      FirecloudMethodQuery newMethodQuery =
+          new FirecloudMethodQuery()
+              .namespace(methodNamespace)
+              .name(methodName)
+              .entityType("Workflow")
+              .snapshotComment(snapshotComment)
+              .payload(sourceFileContents);
+
+      FirecloudMethodResponse methodResponse;
+      if (existingMethods.isEmpty()) {
+        log.info("No existing methods found with given namespace/name; creating new method.");
+        try {
+          methodResponse = apiClientFactory.methodRepositoryApi().createMethod(newMethodQuery);
+        } catch (ApiException e) {
+          log.warning(e.getResponseBody());
+          throw e;
+        }
+      } else {
+        log.info("Method already exists. Creating new snapshot.");
+        int latestSnapshotId =
+            existingMethods.stream()
+                .map(method -> method.getSnapshotId())
+                .max(Integer::compare)
+                .get();
+
+        try {
+          methodResponse =
+              apiClientFactory
+                  .methodRepositoryApi()
+                  .createMethodSnapshot(
+                      methodNamespace,
+                      methodName,
+                      Integer.toString(latestSnapshotId),
```

Comment:
Hm.. this is confusing. This is the snapshot ID of the most recent, not the one that's being created right? What would be the significance of using an older snapshot ID here?